crates/red_knot_python_semantic/src/types.rs

```diff
@@ -6138,6 +6138,24 @@ impl<'db> FunctionType<'db> {
         Type::BoundMethod(BoundMethodType::new(db, self, self_instance))
     }
 
+    /// Returns the AST node for this function.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the file passed in does not match the one where this function is defined.
+    pub(crate) fn node(self, db: &'db dyn Db, file: File) -> &'db ast::StmtFunctionDef {
+        let body_scope = self.body_scope(db);
+
+        assert_eq!(
```

Comment:
nit: I'd probably make this just a `debug_assert_eq!`, we probably don't need to pay for this check in release builds, it should be an issue that is caught in dev